nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -488,7 +488,9 @@ in
                       extraConfig = { options, ... }: {
                         _file = "module at ${__curPos.file}:${toString __curPos.line}";
                         config = {
-                          nixpkgs = if options.nixpkgs?hostPlatform && host.options.nixpkgs.hostPlatform.isDefined
```

Comment:
Would be nice modernize this for the case where `pkgs` is set.

```diff
-nixpkgs = if options.nixpkgs?hostPlatform && host.options.nixpkgs.hostPlatform.isDefined
+nixpkgs = if options.nixpkgs?hostPlatform && (host.options.nixpkgs.hostPlatform.isDefined || host.options.nixpkgs.pkgs.isDefined)
```

This is not necessary for solving your problem, but using the platform flavored options is more consistent and should allow e.g. to declare a cross compiled container without using `mkForce` in `nixpkgs.*` - just set `buildPlatform`.

Even nicer would be `!host.options.nixpkgs.system.isDefined` but it's not that simple. It's always defined, so it needs to be a `highestPrio` comparison. Also the other ones would need to be checked. It's complicated.